### PR TITLE
PHOENIX-5826 Remove guava from queryserver

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

**Comment by stoty** [comment]
The non-trivial parts are:

Copying HostAndPort with minimal changes, to ensure backward compatibility in the ZK string format

The SimpleLRUCache implementation that is used instead of Guava cache.

File: queryserver/src/main/java/org/apache/phoenix/util/HostAndPort.java [hunk 1]
```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (C) 2011 The Guava Authors
```
Comment (joshelser) [inline queryserver/src/main/java/org/apache/phoenix/util/HostAndPort.java:2]: We should ensure that a top-level NOTICE file propagates this copyright notice, as well as the containing jar.

the ASF parent pom already has convention to append entries to the NOTICE file. One example https://github.com/apache/hbase/blob/master/hbase-thrift/src/main/appended-resources/META-INF/NOTICE. The "src/main/appended-resources" is the magic set up by asf.pom.

File: queryserver/src/main/java/org/apache/phoenix/util/HostAndPort.java [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+ *   <li>[2001:db8::1]:80  - {@link #getHostText()} omits brackets
+ *   <li>2001:db8::1       - Use {@link #requireBracketsForIPv6()} to prohibit this
+ * </ul>
+ *
+ * <p>Note that this is not an exhaustive list, because these methods are only
+ * concerned with brackets, colons, and port numbers.  Full validation of the
+ * host field (if desired) is the caller's responsibility.
+ *
+ * @author Paul Marks
+ * @since 10.0
```
Comment (joshelser) [inline queryserver/src/main/java/org/apache/phoenix/util/HostAndPort.java:55]: Can we include a comment somewhere in here which states what version of Guava this was copied from?

File: queryserver/src/main/java/org/apache/phoenix/util/SimpleLRUCache.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            if (currentSize <= triggerSize) {
+                return;
+            }
+            LOG.warn("UGI Cache capacity exceeded, you may want to increase its size");
+            TreeSet<Entry<Object, AtomicLong>> sortedByLRU = new TreeSet<>(
+                    new Comparator<Entry<Object, AtomicLong>>() {
+                        @Override
+                        public int compare(Entry<Object, AtomicLong> o1, 
+                                Entry<Object, AtomicLong> o2) {
+                            if( Long.compare(o2.getValue().get(), o1.getValue().get()) == 0) {
```
Comment (joshelser) [inline queryserver/src/main/java/org/apache/phoenix/util/SimpleLRUCache.java:78]: Compute this once, and use it for your `if` and `return`

File: queryserver/src/main/java/org/apache/phoenix/util/HostAndPort.java (hunk 2 above)
Comment (stoty) [reply queryserver/src/main/java/org/apache/phoenix/util/HostAndPort.java:55]: Yes, done.

File: queryserver/src/main/java/org/apache/phoenix/util/SimpleLRUCache.java (hunk 3 above)
Comment (stoty) [reply queryserver/src/main/java/org/apache/phoenix/util/SimpleLRUCache.java:78]: Ok, done.

**Comment by stoty** [comment]
Making sure that the licensing documentation is in order seems to be a bigger task than the whole Guava removal.

I have split that out into https://issues.apache.org/jira/browse/PHOENIX-5901

**Comment by stoty** [comment]
I forgot to actually push the changes. Better late than never.

**Comment by stoty** [comment]
Thank you @joshelser .
I wasn't comfortable with the comparator returning inconsistent results when the timestamp is equal, so I've changed the implementation a bit.
Could you re-check the last commit ?

**Comment by joshelser** [comment]
> Could you re-check the last commit ?

Looks fine. Thanks for double checking.

**Pull request closed without merging**